src/core/MythosArea.ttslua

```diff
@@ -26,22 +26,51 @@ local currentScenario, useFrontData, tokenData
 
 function onSave()
   return JSON.encode({
-    currentScenario = currentScenario,
-    useFrontData    = useFrontData,
-    tokenData       = tokenData
+    scenarioCardGUID  = scenarioCardGUID,
+    currentScenario   = currentScenario,
+    useFrontData      = useFrontData,
+    tokenData         = tokenData
   })
 end
 
 function onLoad(savedData)
   if savedData and savedData ~= "" then
     local loadedState = JSON.decode(savedData) or {}
+    scenarioCard      = getObjectFromGUID(loadedState.scenarioCardGUID) or nil
     currentScenario   = loadedState.currentScenario or ""
     useFrontData      = loadedState.useFrontData or true
     tokenData         = loadedState.tokenData or {}
   end
   Wait.time(function() collisionEnabled = true end, 0.1)
 end
 
+function copyScenarioReferenceCard()
+  if scenarioCard == nil then broadcastToAll("No scenario reference card found.", "Red") return end
+  
+  local usedColors = playermatApi.getUsedMatColors()
+  if #usedColors == 0 then broadcastToAll("No investigators placed.", "Red") return end
+  for _, color in ipairs(usedColors) do
+    local cardPosition = playermatApi.transformLocalPosition({-1.365, 0.1, -0.625}, color)
+    local searchResult = searchLib.atPosition(cardPosition, "isCardOrDeck")
+
+    if #searchResult > 1 then goto nextLoop end
+    if #searchResult == 1 then
+      local obj = searchResult[1]
+      local md = JSON.decode(obj.getGMNotes()) or {}
+
+      if md.type == "ScenarioReference" then
+        local matColor = playermatApi.getMatColorByPosition(obj.getPosition())
+        local ownedObjects = guidReferenceApi.getObjectsByOwner(matColor)
+        ownedObjects.Trash.putObject(obj)
+      else
+        goto nextLoop
+      end
+    end
+    scenarioCard.clone({position = cardPosition, snap_to_grid = true})
+    ::nextLoop::
```

Comment:
You can simplify this section (this will replace LOC 56 to 71):
```lua
if #searchResult == 1 then
  local obj = searchResult[1]
  local md = JSON.decode(obj.getGMNotes()) or {}

  if md.type == "ScenarioReference" then
    local trash = guidReferenceApi.getObjectByOwnerAndType(color, "Trash")
    trash.putObject(obj)
    scenarioCard.clone({position = cardPosition, snap_to_grid = true})
  end
end
```